### ISSUE TITLE
Ansible inventory loading targets incorrect hosts

### ISSUE THREAD

**Issue by retr0h** [philpep/testinfra#347]
Molecule writes out an Ansible inventory file using the yaml format.  We then invoke the testinfra verifier and pass the yaml inventory file.

The testinfra tests are written and rely [testinfra_hosts](https://github.com/retr0h/molecule-vagrant-bug/blob/master/molecule/default/tests/test_default.py#L5-L6).  However, when doing this testinfra targets the wrong hosts when using 'all'.

This can be reproduced by:

* Checkout https://github.com/retr0h/molecule-vagrant-bug
* cd into the checkout
* `pip install -r bad-requirements.txt`
* run `molecule create` (on a system that has virtualbox, vagrant)
* run `molecule verify`

You will see the following:

```
============================= test session starts ==============================
    platform darwin -- Python 2.7.13, pytest-3.6.3, py-1.5.4, pluggy-0.6.0 -- /Users/jodewey/git/molecule-vagrant-bug/venv/bin/python2.7
    rootdir: /Users/jodewey/git/molecule-vagrant-bug/molecule/default, inifile:
    plugins: testinfra-1.13.1
collected 2 items

    tests/test_default.py::test_hosts_file[ansible://ubuntu-bionic64] ubuntu-xenial64
    PASSED
    tests/test_default.py::test_hosts_file[ansible://ubuntu-xenial64] ubuntu-bionic64
    PASSED

=========================== 2 passed in 1.93 seconds ===========================
```

As you see the Testinfra connection URL (which testinfra constructs) does not match the `hostname` print output from the test.

This problem started with Testinfra 1.13.1.  It appears to be fine in 1.13.0.

```
pip install -r working-requirements.txt
molecule verify

============================= test session starts ==============================
    platform darwin -- Python 2.7.13, pytest-3.6.3, py-1.5.4, pluggy-0.6.0 -- /Users/jodewey/git/molecule-vagrant-bug/venv/bin/python2.7
    rootdir: /Users/jodewey/git/molecule-vagrant-bug/molecule/default, inifile:
    plugins: testinfra-1.13.0
collected 2 items

    tests/test_default.py::test_hosts_file[ansible://ubuntu-xenial64] ubuntu-xenial64
    PASSED
    tests/test_default.py::test_hosts_file[ansible://ubuntu-bionic64] ubuntu-bionic64
    PASSED
=========================== 2 passed in 2.15 seconds ===========================
```

**Comment by philpep** [comment]
Hi, thanks for reporting.
If this has been introduced in 1.13.1 I think this might be related with febe2cc
We also have a PR which seems related to this issue: https://github.com/philpep/testinfra/pull/343

I'll dig in the bug ASAP to provide the proper fix. thanks.

**Comment by philpep** [comment]
This should be fixed in 1.14.1 thanks!
